api/src/services/script-runner.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { Request as ExpressRequest } from 'express';
+import { PrismaService } from './prisma.service';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { User } from '../dtos/users/user.dto';
+import { mapTo } from '../utilities/mapTo';
+
+/**
+  this is the service for running scripts
+  most functions in here will be unique, but each function should only be allowed to fire once
+*/
+@Injectable()
+export class ScriptRunnerService {
+  constructor(private prisma: PrismaService) {}
+
+  /**
+    this is simply and example
+  */
+  async example(req: ExpressRequest): Promise<SuccessDTO> {
+    const requestingUser = mapTo(User, req['user']);
+    await this.markScriptAsRunStart('example', requestingUser);
+    const rawJurisdictions = await this.prisma.jurisdictions.findMany();
+    await this.markScriptAsComplete('example', requestingUser);
+    return { success: !!rawJurisdictions.length };
+  }
+
+  // |------------------ HELPERS GO BELOW ------------------ | //
+
+  /**
+   *
+   * @param scriptName the name of the script that is going to be run
+   * @param userTriggeringTheRun the user that is attempting to trigger the script run
+   * @description this checks to see if the script has already ran, if not marks the script in the db
```

Comment:
What if we intentionally want a script to be run multiple times?

Reply:
I think we should put some friction around running a script multiple times

we always have the option of deleting the script run entry in the db and running again. But I think most scripts should be considered fire once

if we really needed to we could do some stuff with creating multiple endpoints and each endpoint sends a unique script name to the same function in the service so I do think that option is still available